File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        if numRows == 1:
            return [[1]]
        if numRows == 2:
            return [[1], [1,1]]
        ans = [[1], [1,1]]
        for i in range(3, numRows+1):
            row = [1]
            for j in range(i-2):
                row.append(ans[i-2][j]+ans[i-2][j+1])
            row.append(1)
            ans.append(row)
        return ans
            
            

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        if (numRows == 1) return {{1}};
        if (numRows == 2) return {{1}, {1,1}};
        vector<vector<int>> ans;
        ans.push_back({1});
        ans.push_back({1,1});
        for (int i = 2; i < numRows; i++) {
            vector<int> temp;
            temp.push_back(1);
            for (int j = 0; j < i - 1; j++) {
                temp.push_back(ans[i-1][j] + ans[i-1][j+1]);
            }
            temp.push_back(1);
            ans.push_back(temp);
        }
        return ans;
    }
};